### PR TITLE
Dev/frame data aug

### PR DESCRIPTION
**Pull request by Purg**
## What does this PR do?
This adds a drop-out augmentation on a window of FrameData structures that would go into vectorization, integrating its use into the TCN dataset as well as training.

Subsequently, this removes pre-vectorization caching as we can no longer sensibly do that with randomized augmentation happening.

### CODE REVIEW

**Comment by Purg** [comment]
I need to make some fixes, some spot checks are failing what the new vectorizer.

**Review by cameron-a-johnson — APPROVED** [review]

File: tcn_hpl/callbacks/plot_metrics.py [hunk 1]
```diff
@@ -248,54 +266,56 @@ def on_validation_epoch_end(
...
-            per_video_frame_gt_preds[source_vid][source_frame] = (int(gt), int(pred))
-
-        plot_gt_vs_preds(self.output_dir, per_video_frame_gt_preds, split="validation")
+        curr_f1 = pl_module.val_f1.compute()
+        best_f1 = pl_module.val_f1_best.compute()
 
         #
         # Create confusion matrix
         #
+        class_ids = np.arange(all_probs.shape[-1])
```
Comment (josephvanpeltkw) [inline tcn_hpl/callbacks/plot_metrics.py:275]: Is this a repeat of line 199?

File: tcn_hpl/callbacks/plot_metrics.py [hunk 2]
```diff
@@ -363,27 +385,28 @@ def on_test_epoch_end(
...
```
Comment (josephvanpeltkw) [inline tcn_hpl/callbacks/plot_metrics.py:388]: another one here - unless the editor on github is hiding more context from me

**Review by josephvanpeltkw — APPROVED** [review]

File: tcn_hpl/callbacks/plot_metrics.py (hunk 1 above)
Comment (Purg) [reply tcn_hpl/callbacks/plot_metrics.py:275]: Yes, but these are different methods that are doing this with different inputs. I'm going to blame github of not showing you enough context.

Comment (Purg) [reply tcn_hpl/callbacks/plot_metrics.py:275]: `def on_train_epoch_end` vs. `def on_validation_epoch_end` vs. `def on_test_epoch_end`

**Pull request merged**